components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -169,9 +170,11 @@ public IdentityProvider getResidentIdP(String tenantDomain)
             openIdUrl = IdentityUtil.getServerURL(IdentityConstants.OpenId.OPENID, true, true);
         }
 
-        samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true, true, false);
+        samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true)
+                + IdPManagementUtil.getTenantParameter();
```

Comment:
why we send a tenant parameter here. What's the URL that gets constructed

Reply:
This was the earlier implementation. Before we used the modified getServerURL method

Reply:
This constructs https://localhost:9443/samlsso?tenantDomain=wso2.com

Reply:
with the new method, we take the constructed url mentioned above, checks if tenantQualifiedURLs is enabled, and if yes, resets it to https://localhost:9443/t/wso2.com/samlsso

Reply:
I see, so this has been something existed. +1 and from the underlying code it